### PR TITLE
human-readable test fixtures

### LINKED ISSUE #10007: fix: human-readable test fixtures -autocreated
This issue was created because no open issues were referenced in your PR (#10006).

### PR DESCRIPTION
**Pull request by djellemah**
### Prepare

- [x] Read [PR guidelines](https://github.com/JanssenProject/jans/blob/main/docs/CONTRIBUTING.md#prs)
- [x] Read [license information](https://github.com/JanssenProject/jans/blob/main/LICENSE)

-------------------

### Description

There are two main goals this PR aims to achieve:

1) Extend the `policy-store_xxx.json` to allow for metadata for `cedar_schema` and `policy_content` fields.

2) Provide a proof of concept for a 100% human-readable test fixture, in yaml.

#### Target issue
  https://github.com/JanssenProject/jans/issues/9961
  
#### Implementation Details

1) In `policy-store_xxx.json` files, allow for `cedar_schema` and `policy_content` fields to store human-readable values, rather than only the base64-encoded values that agama lab produces. To achieve this, the `policy-store_xxx.json` json for `cedar_schema` and `policy_content` is extended slightly with the following:

```
"policy_content": {
    "encoding": "none",
    "content_type": "cedar",
    "body": "permit(...) when { ... };"
}
```

- `encoding` can be one of `none` or `base64`

- `content_type` can be one of `cedar` or `cedar-json` for `cedar_schema`, but only `cedar` for `policy_content`

- `body` must contain a value that conforms with the above two.

Where the value for `cedar_schema` and `policy_content` is a plain string, it is assumed that the current convention is used:

- for `cedar_schema`, the value is base64 encoded, and contains the cedar-json representation of a schema

- for `policy_content`, the value is base64 encoded, and contains the cedar representation of a policy

2) This is a proof-of-concept, and intended for test fixtures only. Since serde is deisnged to deserialize from several formats, and since `serde_yml` exists, and since json is a proper subset of yaml, the last commit contains a 100% human readable test fixture containing cedar embedded in yaml.

-------------------
### Test and Document the changes
- [x] Static code analysis has been run locally and issues have been fixed
- [x] Relevant unit and integration tests have been added/updated

NOTE that docs will be updated once it's been agreed that the above code changes make sense.

- [ ] Relevant documentation has been updated if any (i.e. user guides, installation and configuration guides, technical design docs etc)

Please check the below before submitting your PR. The PR will not be merged if there are no commits that start with `docs:` to indicate documentation changes or if the below checklist is not selected.
- [ ] **I confirm that there is no impact on the docs due to the code changes in this PR.**

Closes #10007,

### CODE REVIEW

**Comment by mo-auto** [comment]
Error: Hi @djellemah, You did not reference an open issue in your [PR](10006). I attempted to create an [issue](#10007) for you.    
 Please update that [issues'](#10007) title and body and make sure I correctly referenced it in the above PRs body.

File: jans-cedarling/cedarling/src/common/cedar_schema.rs [hunk 1]
```diff
@@ -6,31 +6,153 @@
...
     pub json: cedar_json::CedarSchemaJson,
 }
 
+impl PartialEq for CedarSchema {
+    fn eq(&self, other: &Self) -> bool {
+        // Have to check principals, resources, action_groups, entity_types,
+        // actions. Those can contain duplicates, and are not stored in comparison order.
+        // So use HashSet to compare them.
+
+        let self_principals = self.schema.principals().collect::<std::collections::HashSet<_>>();
```
Comment (olehbozhok) [inline jans-cedarling/cedarling/src/common/cedar_schema.rs]: Why you not import HashSet?
Do you like long form usage `std::collections::HashSet`?

**Review by olehbozhok — DISMISSED** [review]
Looks OK to me

**Review by rmarinn — DISMISSED** [review]
looks good to me but i'm curious why you chose YAML over TOML since Cargo prefers using TOML. But honestly, i think it doesn't matter too much in the end 😅

**Review by duttarnab — APPROVED** [review]

File: jans-cedarling/cedarling/src/common/cedar_schema.rs (hunk 1 above)
Comment (djellemah) [reply jans-cedarling/cedarling/src/common/cedar_schema.rs]: I try to keep `use` as local as possible. Which I know is a little uncommon.

But thanks - this one was a cut'n'paste and should be at least local to the function.

**Comment by djellemah** [comment]
> looks good to me but i'm curious why you chose YAML over TOML since Cargo prefers using TOML. But honestly, i think it doesn't matter too much in the end 😅

Well, two things really:

1) YAML has excellent support for embedded, indented, multiline values. Which is exactly what we need for cedar.

2) It supports anything that `serde` does, so my guess is supporting TOML would be almost trivial. Does TOML support multiline strings? I've never really used it outside the context of config files.

**Review by olehbozhok — APPROVED** [review]

**Review by nynymike — APPROVED** [review]

**Pull request merged**